csrc/fusion_segmenter.cpp

```diff
@@ -1861,7 +1882,12 @@ std::unique_ptr<Fusion> SegmentedFusion::makeFusion(SegmentedGroup* sg) {
   // new Vals so that they can be bound to the segment inputs.
   convertInputRfactorsToRoots(fusion_segment.get());
 
-  return fusion_segment;
+  return std::make_pair(complete_to_segment_map, std::move(fusion_segment));
+}
+
+std::unique_ptr<Fusion> SegmentedFusion::makeFusion(SegmentedGroup* sg) {
```

Comment:
Nit: Given `makeFusion` is not any faster than `makeFusionWithCloner`, I'd just keep one version, name it `makeFusion` as is, and let it always return a pair. The user can easily 

```
auto&& [ir_cloner, fusion] = makeFusion(sg);
```

or 

```
makeFusion(sg)[1];
```

This avoids an unnecessary overloading, and makes it clear to the user that `makeFusion` creates an IrCloner even if it's not used.